cmake/OpenCVFindLibsPerf.cmake

```diff
@@ -39,6 +39,34 @@ if(WITH_IPP)
 endif()
 
 # --- CUDA ---
+# Function to calculate the version range based on match_minor flag
+function(get_version_range version match_minor lower_bound upper_bound)
+    # Split the version into components
+    string(REPLACE "." ";" version_list ${version})
+    list(LENGTH version_list version_length)
+
+    # Extract major and minor components
+    list(GET version_list 0 major)
+    if(version_length GREATER 1)
+        list(GET version_list 1 minor)
+    else()
+        set(minor 0)  # Default minor to 0 if not provided
+    endif()
+
+    # Determine the range based on match_minor flag
+    if(match_minor)
+        # Match minor: range is [major.minor, major.<minor + 1>)
+        math(EXPR new_minor "${minor} + 1")
+        set(${lower_bound} "${major}.${minor}" PARENT_SCOPE)
+        set(${upper_bound} "${major}.${new_minor}" PARENT_SCOPE)
+    else()
+        # Do not match minor: range is [major, <major + 1>)
+        math(EXPR new_major "${major} + 1")
+        set(${lower_bound} "${major}" PARENT_SCOPE)
+        set(${upper_bound} "${new_major}" PARENT_SCOPE)
+    endif()
```

Comment:
I propose for the case when `match_minor` is off use range major.minor...major+1.0. NVidia may introduce backward compatible change, but not forward compatible.

Reply:
that is why the match_minor by default is ON. But the user can choose to ignore minor version.

Personally, I never encounter API incompatibility during minor version update. The case the application breaks due to cuda toolkit update is the PTX, which is actually a problem of system maintenance: you simply should not update cuda toolkit without update driver.

Other kind of breakage is due to bugs introduced in each minor version, but then it is not API breakage, the library still be able to link to whatever minor version. If one knows some version give out wrong executing result they on should just choose a different version when he compile his code, as library provider we should not care.